eng/tools/azure-sdk-tools/azpysdk/sphinx.py

```diff
@@ -0,0 +1,335 @@
+            )
+        )
+        return 1
+    return 0
+
+# build helper functions
+def move_output_and_compress(target_dir: str, package_dir: str, package_name: str) -> None:
+    if not os.path.exists(ci_doc_dir):
+        os.mkdir(ci_doc_dir)
+
+    individual_zip_location = os.path.join(ci_doc_dir, package_dir, package_name)
+    shutil.make_archive(individual_zip_location, 'gztar', target_dir)
+
+def should_build_docs(package_name: str) -> bool:
+    return not ("nspkg" in package_name or package_name in ["azure", "azure-mgmt", "azure-keyvault", "azure-documentdb", "azure-mgmt-documentdb", "azure-servicemanagement-legacy", "azure-core-tracing-opencensus"])
+
+def sphinx_build(package_dir: str, target_dir: str, output_dir: str, fail_on_warning: bool) -> int:
+    command_array = [
+                "sphinx-build",
+                "-b",
+                "html",
+                "-A",
+                "include_index_link=True",
+                "-c",
+                sphinx_conf_dir,
+                target_dir,
+                output_dir
+            ]
+    if fail_on_warning:
+        command_array.append("-W")
+        command_array.append("--keep-going")
+
+    try:
+        logger.info("Sphinx build command: {}".format(command_array))
+        check_call(
+            command_array,
+            cwd=package_dir
+        )
+    except CalledProcessError as e:
+        logger.error(
+            "sphinx-build failed for path {} exited with error {}".format(
+                target_dir, e.returncode
+            )
+        )
+        if in_analyze_weekly():
+            from gh_tools.vnext_issue_creator import create_vnext_issue
+            create_vnext_issue(package_dir, "sphinx")
+        return 1
+    return 0
+
+class sphinx(Check):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def register(self, subparsers: "argparse._SubParsersAction", parent_parsers: Optional[List[argparse.ArgumentParser]] = None) -> None:
+        """Register the `sphinx` check. The sphinx check installs sphinx and and builds sphinx documentation for the target package.
+        """
+        parents = parent_parsers or []
+        p = subparsers.add_parser("sphinx", parents=parents, help="Prepares a doc folder for consumption by sphinx, runs sphinx-apidoc against target folder and handles management generation, and run sphinx-build against target folder. Zips and moves resulting files to a root location as well.")
+        p.set_defaults(func=self.run)
+
+        p.add_argument(
+            "--next",
+            default=False,
+            help="Next version of sphinx is being tested",
+            required=False
+        )
+
+        p.add_argument(
+            "--inci",
+            dest="in_ci",
+            action="store_true",
+            default=False
+        )
+
+    def run(self, args: argparse.Namespace) -> int:
+        """Run the sphinx check command."""
+        logger.info("Running sphinx check in isolated venv...")
```

Comment:
Nit: if we're going to have this and also say `isolated` while still only isolating based on `args.isolate`, I don't really agree with the log message 👍 

We could FORCE it to always isolate by changing the call to `self.get_executable(True,...)`